tool2schema/config.py

```diff
@@ -0,0 +1,52 @@
+from __future__ import annotations
+
+from typing import Optional
+
+
+class Config:
+    """
+    Configuration class for tool2schema.
+    """
+
+    def __init__(self, parent: Optional[Config] = None, **settings):
+        self._parent = parent
+        self._settings = settings
+
+    @property
+    def ignore_parameters(self) -> list[str]:
+        """
+        List of parameter names to ignore when creating a schema.
+        """
+        return self._get_setting(Config.ignore_parameters.fget.__name__, ["self", "args", "kwargs"])
+
+    @ignore_parameters.setter
+    def ignore_parameters(self, value: list[str]):
+        self._set_setting(Config.ignore_parameters.fget.__name__, value)
+
+    def reset_default(self):
+        """
+        Reset the configuration to the default settings.
+        """
+        self._settings = {}
```

Comment:
Should we reset to `{}` or to the `settings` argument in the constructor?

Reply:
It depends, do we want it to reset everything or reset it to the settings given in the constructor?
```python
config = Config(ignore_parameters=True)
config.ignore_function_description=True
config.reset_default() # should this revert only 'ignore_function_description' or also 'ignore_parameters'?
```
In the case of the global configuration it makes no difference

Reply:
To me, only reverting `ignore_function_description` makes sense to me.

Reply:
Done with 5a6b214, added two tests to check this